src/baldr/datetime.cc

```diff
@@ -51,8 +51,9 @@ namespace DateTime {
 tz_db_t::tz_db_t() : db(date::get_tzdb()) {
   // NOTE: outside of this class 0 is reserved for invalid timezone
   // so we offset each index by 1 to get into the valid range 1-300 or so
-  for (size_t i = 0; i < db.zones.size(); ++i) {
-    names.emplace(db.zones[i].name(), i + 1);
+  size_t idx{0};
```

Comment:
im not sure how i feel about this. sometimes i do this too but most times i question it. i know we like the `foreach` because it reads easier but we need the index incremented anyway. then my mind says if you are in-the-know enough to understand the difference between pre and post increment operators you can probably handle an index based for loop. anyway i could go either way on it, but imho its not a necessary change

Reply:
sorry i didn't explain myself very well. my point was that range loops are syntax sugar for readability when you dont need the index. i am perfectly aware of the pre-increment operator im just saying that it is another detail to know about which is just as mentally demanding as reading an index-based for loop. i agree that in this case the final result might be faster because there wont be any calls to `operator[]` (though maybe the compiler can figure out how to do this properly with pointer arithmetic, ie the iterator version which would be equivalent).

anyway its a fine change, i was just trying to explain my hesitation to a change that seems unrelated to warnings and modernization. regarding the latter, index based for loops arent deprecated or anything so we dont need to eg. remove all of them from the code base